### PR TITLE
Don't rotate static game tiles and fix switch rotation

### PR DESCRIPTION
**Pull request by 12pm**
Should be handy for mapping.
Can be disabled by activating unused tiles (don't think it's worth a separate toggle and if you want to do it, you're doing something unusal)

### CODE REVIEW

File: src/game/mapitems.cpp [hunk 1]
```diff
@@ -91,3 +91,18 @@ bool IsValidEntity(int Index)
 		||  Index == ENTITY_DOOR
 	);
 }
+
+bool IsRotatableTile(int Index)
+{
+	return (
+			Index == TILE_STOP
```
Comment (def-) [inline src/game/mapitems.cpp]: Indent the rest of this with spaces (2 tabs, 4 spaces I think)

File: src/game/editor/layer_tiles.cpp [hunk 2]
```diff
@@ -477,13 +485,19 @@ void CLayerTiles::BrushRotate(float Amount)
...
 		CTile *pDst = m_pTiles;
+		bool Rotate = !(m_Game || m_Front || m_Tele || m_Switch || m_Tune || m_Speedup) || m_pEditor->m_AllowPlaceUnusedTiles;
 		for(int x = 0; x < m_Width; ++x)
 			for(int y = m_Height-1; y >= 0; --y, ++pDst)
 			{
 				*pDst = pTempData[y*m_Width+x];
-				if(pDst->m_Flags&TILEFLAG_ROTATE)
-					pDst->m_Flags ^= (TILEFLAG_HFLIP|TILEFLAG_VFLIP);
-				pDst->m_Flags ^= TILEFLAG_ROTATE;
+				if(!Rotate && !IsRotatableTile(pDst->m_Index))
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp]: Currently m_AllowPlaceUnusedTiles can't overwrite if !IsRotatableTile, right? Maybe move the m_AllowPlaceUnusedTiles check down here?

File: src/game/editor/layer_tiles.cpp [hunk 3]
```diff
@@ -477,13 +485,19 @@ void CLayerTiles::BrushRotate(float Amount)
 		CTile *pTempData = new CTile[m_Width*m_Height];
 		mem_copy(pTempData, m_pTiles, m_Width*m_Height*sizeof(CTile));
 		CTile *pDst = m_pTiles;
+		bool Rotate = !(m_Game || m_Front || m_Tele || m_Switch || m_Tune || m_Speedup) || m_pEditor->m_AllowPlaceUnusedTiles;
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp]: This check is duplicated a few times, might also move `(m_Game || m_Front || m_Tele || m_Switch || m_Tune || m_Speedup)` out to `IsRotatableLayer()`

File: src/game/editor/layer_tiles.cpp [hunk 4]
```diff
@@ -1654,6 +1668,68 @@ void CLayerSwitch::BrushDraw(CLayer *pBrush, float wx, float wy)
 	FlagModified(sx, sy, l->m_Width, l->m_Height);
 }
 
+void CLayerSwitch::BrushFlipX()
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp]: Why is there this special handling for switches?

File: src/game/editor/layer_tiles.cpp [hunk 5]
```diff
@@ -1654,6 +1668,68 @@ void CLayerSwitch::BrushDraw(CLayer *pBrush, float wx, float wy)
...
+		{
+			CSwitchTile Tmp = m_pSwitchTile[y*m_Width+x];
+			m_pSwitchTile[y*m_Width+x] = m_pSwitchTile[(m_Height-1-y)*m_Width+x];
+			m_pSwitchTile[(m_Height-1-y)*m_Width+x] = Tmp;
+		}
+}
+
+void CLayerSwitch::BrushRotate(float Amount)
+{
+	int Rotation = (round_to_int(360.0f*Amount/(pi*2))/90)%4;	// 0=0°, 1=90°, 2=180°, 3=270°
```
Comment (def-) [inline src/game/editor/layer_tiles.cpp]: Can't reuse CLayerTiles::BrushRotate here?

File: src/game/editor/layer_tiles.cpp (hunk 4 above)
Comment (12pm) [reply src/game/editor/layer_tiles.cpp]: Every layer has its own tiles, so also a function to rotate those. Was just missing for switch layer.

File: src/game/editor/layer_tiles.cpp (hunk 2 above)
Comment (12pm) [reply src/game/editor/layer_tiles.cpp]: If m_AllowPlaceUnusedTiles is true, Rotate is false, so the condition isn't true. Only need to check it once.

File: src/game/editor/layer_tiles.cpp (hunk 5 above)
Comment (12pm) [reply src/game/editor/layer_tiles.cpp]: Hm don't think so

**Comment by 12pm** [comment]
I realized that currently tele, switch, tune and speedup tiles can't even be rotated.. should I leave it that way and make rotating them impossible, or add rotation flags to them?

**Comment by def-** [comment]
What good would it be to rotate those?

**Comment by fokkonaut** [comment]
It was intended NOT to rotate those tiles. Why would you add it now?

**Pull request merged**